nikola/plugins/task/categories.py

```diff
@@ -0,0 +1,172 @@
+# PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS
+# OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+# OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+# SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+"""Render the category pages and feeds."""
+
+from __future__ import unicode_literals
+try:
+    from urlparse import urljoin
+except ImportError:
+    from urllib.parse import urljoin  # NOQA
+
+from nikola.plugin_categories import Taxonomy
+from nikola import utils
+
+
+class ClassifyCategories(Taxonomy):
+    """Classify the posts by categories."""
+
+    name = "classify_categories"
+
+    classification_name = "category"
+    metadata_name = "category"
+    overview_page_variable_name = "categories"
+    overview_page_items_variable_name = "cat_items"
+    overview_page_hierarchy_variable_name = "cat_hierarchy"
+    more_than_one_classifications_per_post = False
+    has_hierarchy = True
+    include_posts_from_subhierarchies = True
+    include_posts_into_hierarchy_root = False
+    show_list_as_subcategories_list = False
+    generate_atom_feeds_for_post_lists = True
+    template_for_classification_overview = "tags.tmpl"
+    add_postcount_in_overview = True
+    always_disable_rss = False
+    apply_to_posts = True
+    apply_to_pages = False
+    minimum_post_count_per_classification_in_overview = 1
+    omit_empty_classifications = True
+    also_create_classifications_from_other_languages = True
+
+    def set_site(self, site):
+        """Set site, which is a Nikola instance."""
+        super(ClassifyCategories, self).set_site(site)
+        self.show_list_as_index = self.site.config['CATEGORY_PAGES_ARE_INDEXES']
+        self.template_for_single_list = "tagindex.tmpl" if self.show_list_as_index else "tag.tmpl"
+
+    def is_enabled(self, lang=None):
+        """Return True if this taxonomy is enabled, or False otherwise."""
+        return True
+
+    def classify(self, post, lang):
+        """Classify the given post for the given language."""
+        cat = post.meta('category', lang=lang).strip()
+        return [cat] if cat else []
+
+    def get_classification_friendly_name(self, classification, lang, only_last_component=False):
+        """Extract a friendly name from the classification."""
+        classification = self.extract_hierarchy(classification)
+        return classification[-1] if classification else ''
+
+    def get_overview_path(self, lang, type='page'):
+        """A path handler for the list of all classifications."""
+        if self.site.config['CATEGORIES_INDEX_PATH'][lang]:
+            return [_f for _f in [self.site.config['CATEGORIES_INDEX_PATH'][lang]] if _f], 'never'
+        else:
+            return [_f for _f in [self.site.config['CATEGORY_PATH'][lang]] if _f], 'always'
+
+    def slugify_tag_name(self, name, lang):
+        """Slugify a tag name."""
+        if lang is None:  # TODO: remove in v8
```

Comment:
No, it already existed in the old tags plugin and also in the new tags plugin, except that it is now duplicated. But yes, calling this specific instance without language is not possible anymore since it is only called from `slugify_category_name`, which sets `lang` to `''` if it was `None`. I'll remove this check.